File: server/syncmed/Entities/Schedulling.cs
namespace syncmed.Entities
{
    public class Schedulling
    {
        protected Schedulling() { }

        public Schedulling(DateTime date, TimeSpan hour, int idPatient, int idProfessional, string? obs)
        {
            Date = date;
            Hour = hour;
            IdPatient = idPatient;
            IdProfessional = idProfessional;
            Obs = obs;
        }

        public int Id { get; private set; }
        public DateTime Date { get; private set; }
        public TimeSpan Hour { get; private set; }
        public int IdPatient { get; private set; }
        public int IdProfessional { get; private set; }
        public string? Obs { get; private set; }
    }
}

File: client/src/services/schedulling.ts
import { Professional, Schedulling, Specialty } from "@/types/api";
import { axiosClient as api } from "./api";

export async function getSchedullingsAllDates(): Promise<string[]> {
  const { data } = await api.get("/Schedulling/scheduled-dates");

  return data;
}

export async function getSchedullingsByDate(
  date: string,
  professionalId?: number,
): Promise<Schedulling[]> {
  const { data } = await api.get("/Schedulling/by-date", {
    params: { date, professionalId },
  });

  return data;
}

export async function postAddSchedulling(
  date: string,
  hour: string,
  idPatient: number,
  idProfessional: number,
  obs: string,
) {
  const { data } = await api.post("/Schedulling", {
    date,
    hour,
    idPatient,
    idProfessional,
    obs,
  });
  return data;
}
